### DISCUSSION TITLE
JIT arbitrary bg colour not working with VueJS

### DISCUSSION THREAD

**Discussion by JiahuiChen99** [tailwindlabs/tailwindcss#7092] [Help] [answered]
**What version of Tailwind CSS are you using?**

"tailwindcss": "^3.0.7"
"postcss": "^8.3.11"

**What build tool (or framework if it abstracts the build tool) are you using?**

"vue": "^3.0.0"

**What version of Node.js are you using?**
v16.13.1

**What browser are you using?**
Chrome

**What operating system are you using?**
Kali Linux

In VueJS **data** I've an array of colours that I'd like to display. However, the JIT arbitrary background colour with interpolation is not working properly. The divs don't get painted. :woman_shrugging: 
![image](https://user-images.githubusercontent.com/37770349/149764974-eb6b220a-757c-4f5b-b778-651a46afe6d4.png)

```
<div class="rounded-full border-2 border-white w-14 h-14"
                     :class="`bg-[${colour}]`"
                     v-for="( colour, index ) in colours" :key="index">
</div>
```

This line is supposed to do the trick, but it doesn't 

```
:class="`bg-[${colour}]`"
```
The hex colours gets interpolated though... I tried to add the arbitrary colours manually and seems to work fine. But it only stays there for the session, once the app is restarted the colours are not there anymore.
![image](https://user-images.githubusercontent.com/37770349/149765461-1a47ebd9-d43a-44dc-9332-02bdf1ab5976.png)

**Comment by samzlab** [comment, marked as answer]
The JIT compiler will not run your javascript code. So what it's seeing is `bg-[${colour}]` which is not recognized as a valid class name. Keep in mind, that the CSS compiling is running at build time and not on the client (what you see in your browser is not that what tailwind is "see" at compile time).

You can solve the problem if you write the full tailwind class names into your `colours` array, like this:

```javascript
colours: [
  'bg-[#264653]',
  'bg-[#2A9D8F]', 
  // etc...
]
```

**Reply by JiahuiChen99** [reply]
I'd have preferred to have the hex colours there, not the entire class name.
Thanks for the help though :+1:

**Reply by aredda** [reply]
Thanks in advance, just wondering, but what if the colors are retrieved from a database? meaning that they can be anything and not limited to the ones in that array.. how would you go around this?

**Reply by matthewhutchings** [reply]
have this issue too. how did you solve?

**Reply by donaldking** [reply]
Same here, any solution yet?

**Reply by JiahuiChen99** [reply]
Guys, what @samzlab said above, makes total sense. No interpolation is allowed, for further information on this see [Dynamic class names](https://tailwindcss.com/docs/content-configuration#dynamic-class-names) section from Tailwind docs. Instead of using dynamic generation just use full names.
